### ISSUE TITLE
Setting of autohide toolbar not logical

### ISSUE THREAD

**Issue by userx-bw** [joewing/jwm#387]
In the xml file here:
 
 Tray x="0" y="2" autohide="off" 

logically to turn it on then one would just replace the word 'off' with 'on' but when doing so jwm throws an warring and does not allow it to work: warring is:

userx%slackwhere ⚡ ~ ⚡> jwm -p
**JWM: warning: /home/userx/.jwmrc[314]: invalid value for autohide: "on"**

**Comment by Israel-** [comment]
Hi @userx-bw That is the old behavior from much earlier (2.2.X era IIRC)
The new current is to hide the tray to a specific direction
```
autohide
Specify the location to hide the tray when not activated. One of left, right, top, bottom, or off. The default is off to disable hiding.
```

**Comment by userx-bw** [comment]
well color me silly on off boolean has been replaced. where is the logic in that? thanks closing this then.

**Comment by Israel-** [comment]
@userx-bw there was an issue about this that required the new syntax... I just don't remember why right now.

**Comment by userx-bw** [comment]
that is all fine and good. but while I got you here, what means would be best to get a hold of you about something else so as not to flood this with it about a way to better exploit a feature in here dealing with your desktop's ability to show a separate image on each desktop. Because the way, or what  I am trying to do it is working but not working perfectly.

**Comment by Israel-** [comment]
Hi @userx-bw I highly suggest reading the documentation thoroughly
http://www.joewing.net/projects/jwm/config.shtml#desktops
Here is how you do it
```xml
<Desktops width="4" height="1">
    <!-- Default for ALL desktops it will show up on the fourth desktop-->
    <Background type="image" >/path/to/picture.ext</Background>
    <!-- specify the first desktop's image here-->
    <Desktop name="1">
         <Background type="image" >/path/to/first/picture.ext</Background>
    </Desktop>
    <Desktop name="2">
         <Background type="image" >/path/to/second/picture.ext</Background>
    </Desktop>
    <Desktop name="3">
         <Background type="image" >/path/to/third/picture.ext</Background>
    </Desktop>
</Desktops>
```
Of course you cannot have something else controlling the background like `pcmanfm` for this to work

**Comment by userx-bw** [comment]
I got that figured out, what I am doing with this feature is I use mhsetroot, a much better hsetroot, if you have heard of hsetroot, I modded it giving it what it really needed. I use it in a script to change images randomly - now it is working but my script I have to fix for pids because it gets called each time I switch to a desktop that is set to call it  

 ```
<Desktop name="1">
         <Background type="command" >/path/to/script/scriptName</Background>
    </Desktop>
```
the only thing I see on "your end" of doing this is that it will take over eventuall and start changing images on desktops that to not have this called to start as well. 

example 
```
  <Background type="image">/usr/share/wallpapers/slack-metal-2560x1600.jpg</Background>
  <!-- spacer -->
  <Desktop>"Main"<Background type="image">/usr/share/wallpapers/slack-metal-2560x1600.jpg</Background></Desktop>
  <Desktop>2<Background type="command">/home/userx/changebgimage</Background></Desktop>
  <Desktop>3<Background type="image">/usr/share/wallpapers/slack-metal-2560x1600.jpg</Background></Desktop>
   <Desktop>4<Background type="command">/home/userx/changebgimage</Background></Desktop>
  <Desktop>5<Background type="image">/home/userx/Pictures/cates-fast-times.jpg</Background></Desktop>
```
that is what mine is right now. If I switch to desktop 2 where that script is called to run then switch back to desktop 1 where an image is called to show, eventually that script takes over that desktop too and started changing images on it instead of it leaving it alone.

**Comment by joewing** [comment]
Auto-hide works the way it is so that JWM doesn't have to guess where you want to hide the tray (where positioning the mouse will cause it to re-appear).  The obvious solution that JWM used to use was, for example, to hide trays at the bottom for horizontal trays positioned at the bottom of the screen, etc.  However, some prefer the tray to hide on the left or right side instead.

As far as desktop backgrounds, I guess I wouldn't recommend mixing having JWM and another program control the background, especially if the other program continues to run as a daemon.  Unless the other program is aware of virtual desktops and the behavior you want, it's going to take over (virtual desktops all share the same root window).  You could use JWM's `command` background to notify another process that the desktop changed, but it would probably be better to just have that program look for the `_NET_CURRENT_DESKTOP` property on the root window.

As far as your tray issue, why do you have `y="1"` instead of `y="0"`?  That should get rid of the gap if I'm understanding correctly.

The `height` attribute in `Desktops` gives you more desktops in another dimension.  Without the pager enabled it's hard to see them, but to access them they are "up" or "down".  There are key bindings you can set to make them available.

**Comment by userx-bw** [comment]
Yeah thanks I've been wondering on the backburner about that how to figure out what desktop - as I am not proficient in C I just figured out how to change that old hsetroot to do what it does now under new name mhsetroot as I went along. for the most part. nevertheless, 

```
userx%slackwhere ⚡ ~ ⚡> [[ $(xprop -root -notype _NET_CURRENT_DESKTOP) -eq '3' ]] && echo "hi"
hi
```

so yeah I think I can work with that  _NET_CURRENT_DESKTOP
as far as the why I didn't use 0 instead of 1 - it was set at -1 so I just said 1 and seen it jump to the top then discovered the crack and tried 2 nope, didn't work and as trying many things at once to set up your WM it just got lost in the crack and I left it like that not moving into figuring that one out as I got stuck on this background image and my script thing.
